src/lib/core/CHIPTLVReader.cpp

```diff
@@ -1573,5 +1573,36 @@ CHIP_ERROR TLVReader::GetNextPacketBuffer(TLVReader & reader, uintptr_t & bufHan
     return CHIP_NO_ERROR;
 }
 
+/**
+ * Get the destination reader with the given tag
```

Comment:
This needs to document the arguments and general functionality.  In particular, at least the following should be documented:

1. What is this assuming about the state of the reader when this function is called?  Presumably that it's positioned right before the first element inside a container?  Or is it meant to be used more generically?  (I know it will do something sane no matter what element it's positioned at; that's an implementation detail and we should document what the intended API use case is.)
2. What does `tagInApiForm` mean?  As a consumer, how do I figure out what to pass in here for a given tag value defined by some TLV-based specification?  Perhaps document that this must be the sort of thing returned by a `ContextTag()` or `CommonTag()` or `ProfileTag()` call?  I wish we had a type for "api tag" that were not a raw integer; that would make all this enforceable by the type system.  Maybe file a followup issue to do that?
3. Is the fact that `destReader` is not modified on failure happenstance or part of the API?

Reply:
And in particular, if the intent is that the argument must be something that was just written into by `OpenContainer`, we should document that.

Reply:
@bzbarsky-apple agree, thank you for your suggestion! just address your comments and update the function description. Also create the tlv tag type ticket https://github.com/project-chip/connectedhomeip/issues/3764